### PR TITLE
Unix: change SetCreationTime to match better with GetCreationTime

### PR DESCRIPTION
**Pull request by tmds**
Fixes https://github.com/dotnet/corefx/issues/33601

CC @danmosemsft @WernerMairl

### CODE REVIEW

**Comment by tmds** [comment]
On fedora, setting mtime updates ctime. From ci test results, this may not be the case on all Linux flavors. I'm going to look into this further.

File: src/System.IO.FileSystem/tests/File/GetSetTimes.cs [hunk 1]
```diff
@@ -34,6 +34,26 @@ public void BirthTimeIsNotNewerThanLowestOfAccessModifiedTimes()
...
+        public void CreationTimeSet_GetReturnsExpected_WhenNotInFuture()
+        {
+            // On Linux, we synthesize CreationTime from the oldest of status changed time (ctime) and write time (mtime).
+            // Changing the CreationTime, updates mtime and causes ctime to change to the current time.
+            // When setting CreationTime to a value that isn't in the future, getting the CreationTime should return the same value.
+
+            string fileName = Path.GetTempFileName();
+            File.WriteAllText(fileName, "");
+
+            FileInfo fileInfo = new System.IO.FileInfo(fileName);
```
Comment (danmosemsft) [inline src/System.IO.FileSystem/tests/File/GetSetTimes.cs]: Don't need System.Io part

File: src/System.IO.FileSystem/tests/File/GetSetTimes.cs [hunk 2]
```diff
@@ -34,6 +34,26 @@ public void BirthTimeIsNotNewerThanLowestOfAccessModifiedTimes()
...
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Linux)]
+        public void CreationTimeSet_GetReturnsExpected_WhenNotInFuture()
+        {
+            // On Linux, we synthesize CreationTime from the oldest of status changed time (ctime) and write time (mtime).
+            // Changing the CreationTime, updates mtime and causes ctime to change to the current time.
+            // When setting CreationTime to a value that isn't in the future, getting the CreationTime should return the same value.
+
```
Comment (danmosemsft) [inline src/System.IO.FileSystem/tests/File/GetSetTimes.cs]: We generally avoid Path.GetTempFileName in favor of using FileCleanupTestBase, which will ensure the file is deleted.

Comment (danmosemsft) [reply src/System.IO.FileSystem/tests/File/GetSetTimes.cs]: Btw I believe GetTempFileName actually creates the file..

Comment (WernerMairl) [reply src/System.IO.FileSystem/tests/File/GetSetTimes.cs]: confirm: GetTempFileName creates file with length 0 - in my personal tests for this i have used a File.Delete after getTempfilename to be sure that i have full control over file creation....

**Comment by tmds** [comment]
> Do we have enough test coverage in file system watcher for this?

@danmosemsft, yes, it is covered by this test:

https://github.com/dotnet/corefx/blob/f149048c3696d1d3342b696f719206fcd1e173cb/src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.File.NotifyFilter.cs#L52-L73

**Comment by tmds** [comment]
I changed the test a bit. Good to merge.

**Comment by tmds** [comment]
Thank you for reviewing @danmosemsft

**Pull request merged**